v1/plugin/stream_collector_proxy.go

```diff
@@ -0,0 +1,126 @@
+package plugin
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/rpc"
+)
+
+type StreamProxy struct {
+	pluginProxy
+
+	plugin StreamCollector
+}
+
+func (c *StreamProxy) StreamMetrics(stream rpc.StreamCollector_StreamMetricsServer) error {
+	// Error channel where we will forward plugin errors to snap where it
+	// can report/handle them.
+	errChan := make(chan string)
+	// Metrics into the plugin from snap.
+	inChan := make(chan []Metric)
+	// Metrics out of the plugin into snap.
+	outChan := make(chan []Metric)
+	err := c.plugin.StreamMetrics(inChan, outChan, errChan)
+	if err != nil {
+		return err
+	}
+	go metricSend(c.plugin, outChan, stream)
+	go errorSend(c.plugin, errChan, stream)
+	streamRecv(c.plugin, inChan, stream)
+	return nil
+}
+
+func errorSend(
+	plugin StreamCollector,
+	ch chan string,
+	s rpc.StreamCollector_StreamMetricsServer) {
+	for r := range ch {
+		reply := &rpc.CollectReply{
+			Error: &rpc.ErrReply{Error: r},
+		}
+		if err := s.Send(reply); err != nil {
+			fmt.Println(err.Error())
```

Comment:
is `fmt.Println..` correct? or should be `log`. I'm not really sure what it is used on this repo ;)